models/evc.py

```diff
@@ -1056,35 +1090,43 @@ def _prepare_push_flow(self, *args, queue_id=None):
         """
         # assign all arguments
         in_interface, out_interface, in_vlan, out_vlan, new_c_vlan = args
-        is_EVPL = (in_vlan is not None)
         flow_mod = self._prepare_flow_mod(
-            in_interface, out_interface, queue_id, is_EVPL
+            in_interface, out_interface, queue_id, in_vlan
         )
-
         # the service tag must be always pushed
         new_action = {"action_type": "set_vlan", "vlan_id": out_vlan}
         flow_mod["actions"].insert(0, new_action)
 
         new_action = {"action_type": "push_vlan", "tag_type": "s"}
         flow_mod["actions"].insert(0, new_action)
 
-        if in_vlan:
+        if in_vlan is not None:
             # if in_vlan is set, it must be included in the match
             flow_mod["match"]["dl_vlan"] = in_vlan
-        if new_c_vlan:
-            # new_in_vlan is set, so an action to set it is necessary
+
+        if new_c_vlan not in {"4096/4096", 0, None}:
```

Comment:
```suggestion
        if new_c_vlan not in self.special_cases:
```

Reply:
Commit [5c98c93c79bffd79edb81fb824d46250e4db683e](https://github.com/kytos-ng/mef_eline/pull/258/commits/5c98c93c79bffd79edb81fb824d46250e4db683e)

Reply:
Looks good, let me close the related threads.